metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/request/SearchRequestHandler.java

```diff
@@ -517,6 +534,23 @@ public static Map<String, Long> extractTermAggregations(@Nonnull SearchResponse
     return extractTermAggregations((ParsedTerms) aggregation, aggregationName.equals("_entityType"));
   }
 
+  private List<SearchSuggestion> extractSearchSuggestions(@Nonnull SearchResponse searchResponse) {
+    final List<SearchSuggestion> searchSuggestions = new ArrayList<>();
+    if (searchResponse.getSuggest() != null) {
+      TermSuggestion termSuggestion = searchResponse.getSuggest().getSuggestion(NAME_SUGGESTION);
+      if (termSuggestion != null && termSuggestion.getEntries().size() > 0) {
+        termSuggestion.getEntries().get(0).getOptions().forEach(suggestOption -> {
```

Comment:
Don't need to iterate over all entries here?

Reply:
yup! we're iterating over all of the suggested options so we can return all of them to the frontend. these options are sorted to be the best match first so right now on the frontend we're just showing the first suggestion. but in the future we may want to show multiple suggestions or something. the API allows for multiple suggestions anyways. most of the time this is just one or two options anyways